### DISCUSSION TITLE
Worcester Bosch Greenstar HIU

### DISCUSSION THREAD

**Discussion by Step4df54** [emsesp/EMS-ESP32#1127] [General]
Hi,

I have a Worcester Bosch Greenstar HIU (Heat Interface Unit).
It has an EMS connection and I have tried connecting it to my BBQKees S32 and it recognises it as a generic boiler along with some data points but there seems to be quite a few datapoints missing.

It's be great to add compatibility for this device but I'm a bit unsure of where to start. Any advice would be much appreciated!

Below are the device details as recognised by EMS-ESP

![1Capture](https://user-images.githubusercontent.com/128062682/225675213-a41125af-20cc-430a-8398-c2dd3d098d75.JPG)

**Comment by bbqkees** [comment]
Ok so basically this is not a boiler but a box with a heat exchanger.
Which data points do you get?

**Reply by Step4df54** [reply]
Hi,

Yes correct that it's not a boiler but it generates domestic hot water instantly and supplies heating water to apartments so in operation it's similar to a combi boiler. But rather than burning gas it gets its energy from a 80°C system that runs around many apartments. It's quite a new product and has only been out a few years.

It uses separate heat exchangers for the domestic hot water and the heating side and the flowrate through those heat exchangers is controlled with 2-port motorized valves.

It finds 23 data points in total, I've included a screenshot of those below.

As an example I can see the heating 2-port valve modulation listed as 'burner current power' and read it's percentage modulation as the heating ramps up.
But I can't seem to see the domestic hot water 2-port valve modulation.
Below is a drawing showing the layout of the HIU and there are a number of different sensors that it would be great to monitor. And it would be very interesting to see if we could adjust temperature setpoints such as heating flow temperature.

Would it help if I were to 'scan devices' and post the results here?

![image](https://user-images.githubusercontent.com/128062682/226719564-8278afd1-7dba-4a75-9a52-fa6507febe38.png)

![image](https://user-images.githubusercontent.com/128062682/226719610-78383f6b-4dfa-4e68-996d-ccbd56844870.png)

**Reply by MichaelDvP** [reply]
The latest dev has this product id in the device library, the boiler should now identified correctly.
The "generic boiler" is a fall back for unknown devices and uses all known entities ((including heatpump).  If values missing we have to extract them from the ems-telegrams.
- update to the latest dev
- let it run for some time (30 min)
- save the system information (help page)
- set system log level to "All", stay on this page
- note the value of the missing parameter, try to set something that the value changes while logging
- save the log after minimum 2-3 minutes logging
- open a issue and attach the system info and the log (as file, not copy&past). Describe how the value has changed while logging.

**Reply by Step4df54** [reply]
Thanks, I have submitted it as an issue.
